packages/valory/skills/dynamic_nft_abci/handlers.py

```diff
@@ -82,6 +82,32 @@ class HttpHandler(BaseHttpHandler):
 
     def setup(self) -> None:
         """Implement the setup."""
+        uri_base_hostname = urlparse(self.context.params.token_uri_base).hostname
+
+        # Route regexes
+        hostname_regex = rf".*({uri_base_hostname}|localhost|127.0.0.1)(:\d+)?"
+        address_regex = r"0x[a-fA-F0-9]{40}"
+        self.handler_url_regex = rf"{hostname_regex}\/.*"
+        metadata_url_regex = rf"{hostname_regex}\/\d+"
+        leaderboard_url_regex = rf"{hostname_regex}\/leaderboard"
+        address_status_url_regex = (
+            rf"{hostname_regex}\/address_status/(?P<address>{address_regex})"
+        )
+        health_url_regex = rf"{hostname_regex}\/healthcheck"
+        link_wallet_url_regex = rf"{hostname_regex}\/link"
+
+        # Routes
+        self.routes = {
+            (HttpMethod.GET.value, HttpMethod.HEAD.value): [
+                (metadata_url_regex, self._handle_get_metadata),
+                (leaderboard_url_regex, self._handle_get_leaderboard),
+                (address_status_url_regex, self._handle_get_address_status),
+                (health_url_regex, self._handle_get_health),
+            ],
+            (HttpMethod.POST.value): [(link_wallet_url_regex, self._handle_post_link)],
+        }
```

Comment:
We build a relationship between url regexes and their corresponding handler method

Reply:
This was inspired by Django's [url dispatcher](https://docs.djangoproject.com/en/1.8/topics/http/urls/#example). As they explain [here](https://stackoverflow.com/questions/33343680/can-a-regular-expression-be-used-as-a-key-in-a-dictionary), it is not recommended to have a dict here.